data/resources.go

```diff
@@ -40,57 +36,35 @@ func (r *ResourceStore) GetResources(ctx context.Context, option Options) (*mode
 	return resources, nil
 }
 
-// getResourcesCount returns the total number of resources stored
-func (r *ResourceStore) getResourcesCount() (int, error) {
-	return 2, nil
-}
-
-func (r *ResourceStore) populateResourceList(resourceList []interface{}) ([]interface{}, error) {
-	topics1 := []string{"a", "b", "c", "d"}
-	topics2 := []string{"a", "b", "e", "f"}
-	dateChanges := []string{"a change_notice", "a previous_date"}
-	tempResource1 := models.Standard{
-		URI:             "/a/temp/uri",
-		URIOld:          "/an/old/uri",
-		ContentType:     "api_dataset_landing_page",
-		CDID:            "ASELECTIONOFNUMBERSANDLETTERS123",
-		DatasetID:       "ASELECTIONOFNUMBERSANDLETTERS456",
-		Edition:         "a temporary edition",
-		MetaDescription: "a temporary description",
-		ReleaseDate:     time.Time{}.UTC(),
-		Summary:         "a temporary summary",
-		Title:           "a temporary title",
-		Topics:          topics1,
-		Language:        "string",
-		Survey:          "string",
-		CanonicalTopic:  "string",
+func (r *ResourceStore) populateItems(ctx context.Context) (items []interface{}, numItems int, err error) {
+	logData := log.Data{}
+	// Get a list of JSON files in the embedded 'json_files' directory.
+	files, err := jsonFiles.ReadDir("json_files")
+	if err != nil {
+		log.Fatal(ctx, "Failed to read json_files directory", err)
 	}
+	for range files {
+		numItems++
+	}
+	items = make([]interface{}, numItems)
 
-	resourceList[0] = tempResource1
+	// Loop through files, read, and unmarshal each JSON file into Go structs.
+	for i := 0; i < numItems; i++ {
+		file := files[i]
+		logData["file"] = file.Name()
+		fileBytes, err := jsonFiles.ReadFile("json_files/" + file.Name())
+		if err != nil {
+			log.Fatal(ctx, "Failed to read file", err, logData)
+		}
 
-	tempResource2 := models.Release{
-		URI:             "/another/temp/uri",
-		URIOld:          "/another/old/uri",
-		ContentType:     "api_dataset_landing_page",
-		CDID:            "ASELECTIONOFNUMBERSANDLETTERS789",
-		DatasetID:       "ASELECTIONOFNUMBERSANDLETTERS101112",
-		Edition:         "another temporary edition",
-		MetaDescription: "another temporary description",
-		ReleaseDate:     time.Time{}.UTC(),
-		Summary:         "another temporary summary",
-		Title:           "another temporary title",
-		Topics:          topics2,
-		Language:        "string",
-		Survey:          "string",
-		CanonicalTopic:  "string",
-		Cancelled:       true,
-		Finalised:       true,
-		Published:       true,
-		DateChanges:     dateChanges,
-		ProvisionalDate: "October-November 2024",
-	}
+		var jsonData map[string]interface{}
+		err = json.Unmarshal(fileBytes, &jsonData)
+		if err != nil {
+			log.Fatal(ctx, "Failed to unmarshal JSON for file: "+file.Name(), err)
+		}
 
-	resourceList[1] = tempResource2
+		items[i] = jsonData
+	}
 
-	return resourceList, nil
+	return items, numItems, nil
```

Comment:
If you make my suggestions, then you can calculate the length here. Although to be honest, is there any need to return the length from this function anyway. You could just return the items and the number is the length of those items anyway. 
```suggestion
	return items, len(items), nil
```